diff --git a/rtloader/common/rtloader_mem.h b/rtloader/common/rtloader_mem.h
index a36d61070..c26f2d730 100644
--- a/rtloader/common/rtloader_mem.h
+++ b/rtloader/common/rtloader_mem.h
@@ -65,9 +65,7 @@ void *_malloc(size_t sz);
 void _free(void *ptr);
 
 #ifdef __cplusplus
-#    ifdef _WIN32
-#        define __THROW
-#    elif __APPLE__
+#    ifndef __GLIBC__
 #        define __THROW
 #    endif
 
diff --git a/rtloader/rtloader/CMakeLists.txt b/rtloader/rtloader/CMakeLists.txt
index 0fd0a2abc..bb39b5d17 100644
--- a/rtloader/rtloader/CMakeLists.txt
+++ b/rtloader/rtloader/CMakeLists.txt
@@ -27,6 +27,12 @@ add_library(datadog-agent-rtloader SHARED
     rtloader.cpp
     ../common/rtloader_mem.c
 )
+
+# execinfo is not POSIX-compliant and some libc may have libexecinfo as a library.
+find_library(EXECINFO execinfo)
+if(EXECINFO)
+    target_link_libraries(datadog-agent-rtloader execinfo)
+endif()
 endif()
 
 set_target_properties(datadog-agent-rtloader PROPERTIES
