diff --git a/tasks/system_probe.py b/tasks/system_probe.py
index e320fa9db..9d264b352 100644
--- a/tasks/system_probe.py
+++ b/tasks/system_probe.py
@@ -312,6 +312,8 @@ def build_object_files(ctx, bundle_ebpf=False):
         "-include {}".format(os.path.join(c_dir, "asm_goto_workaround.h")),
         '-O2',
         '-emit-llvm',
+        # Some linux distribution enables stack protector by default which is not available on eBPF
+        '-fno-stack-protector',
     ]
 
     # Mapping used by the kernel, from https://elixir.bootlin.com/linux/latest/source/scripts/subarch.include
